geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/MessageDispatcher.java

```diff
@@ -533,21 +533,28 @@ private boolean handleAuthenticationExpiredException(AuthenticationExpiredExcept
       throws InterruptedException {
     long reAuthenticateWaitTime =
         getSystemProperty(RE_AUTHENTICATE_WAIT_TIME, DEFAULT_RE_AUTHENTICATE_WAIT_TIME);
+    // for old client, don't wait for re-auth but unregister this proxy completely.
+    if (getProxy().getVersion().isOlderThan(RE_AUTHENTICATION_START_VERSION)) {
```

Comment:
This block feels like it should be extracted into it's own method for readability. It distracts from the primary method purpose.

Reply:
Any suggestions on the method name? I thought I will just ask now before I revise the PR.

Reply:
Something like:

```java
if (handledUnsupportedClient(...)) {
  return true;
}
```